packages/xstate-store/src/types.ts

```diff
@@ -113,6 +113,24 @@ export interface Store<
   };
 }
 
+export type StoreConfig<
+  TContext extends StoreContext,
+  TEventPayloadMap extends EventPayloadMap,
+  TEmitted extends EventPayloadMap
+> = {
+  context: TContext;
+  emits?: {
+    [K in keyof TEmitted & string]: (payload: TEmitted[K]) => void;
+  };
+  on: {
+    [K in keyof TEventPayloadMap & string]: StoreAssigner<
+      NoInfer<TContext>,
```

Comment:
I'm guilty as charged but on principle `NoInfer` **shouldn't** be used within type definitions. It's a signature-level "annotation". But ye, well - I see how this type has both an inferrable and non-inferrable usage of `TContext` and having it here makes this reusable... so perhaps it's fine. I know of situations in which this kind of a thing creates problems though. David from `arktype` can confirm 😉. Our usage might be different from `arktype` though so we might be not as susceptible to those problems

Reply:
I'll try without it